talk/morelanguage/exceptions.tex

```diff
@@ -4,185 +4,203 @@
+      \item however, in a tight, performance-critical loop, where the error is handled close to the place it happens a more explicit error code or object might be better
       \begin{itemize}
-      \item one cannot ``template'' the throw clause
+        \item but profile if you really want to find out!
       \end{itemize}
-    \end{itemize}
-  \end{alertblock}
-\end{frame}
-
-\begin{frame}[fragile]
-  \frametitlecpp[11]{What remains}
-  \begin{block}{throw is dead}
-    \begin{itemize}
-    \item dynamic exception specification are deprecated
-    \item even throw() (no exceptions)
-    \end{itemize}
+   \end{itemize}
   \end{block}
-  \pause
-  \begin{exampleblock}{long live noexcept}
-    \begin{itemize}
-    \item noexcept a somehow equivalent to throw()
-    \item but is checked at compile time
-    \item so allows compiler optimizations
-    \end{itemize}
-  \end{exampleblock}
+  \begin{multicols}{2}
+    \begin{minipage}{4.5cm}
+      \begin{alertblock}{Don't}
+        \begin{cppcode*}{fontsize=\tiny,gobble=2,linenos=false}
+          // try ... catch in hot loop might
+          //    impact performance
+          for (const &item: list) {
+            try {
+              check_item(item); // Can throw
+              ... // process item
+            }
+            catch (bad_item &e) {
+              ... // ignore and keep going
+            }
+          }
+        \end{cppcode*}
+      \end{alertblock}
+    \end{minipage}
+    \columnbreak
+    \begin{minipage}{5.8cm}
+      \begin{exampleblock}{Do}
+        \begin{cppcode*}{fontsize=\tiny,gobble=2,linenos=false}
+          // For this code, explicit error codes
+          //   may be better
+          for (const &item: list) {
+            std::optional<T> err_code = check_item(item);
+            if (!err_code) {
+              ... // process item
+            } else {
+              ... // warn and carry on
+            }
+          }
+        \end{cppcode*}
+      \end{exampleblock}
+    \end{minipage}
+  \end{multicols}
 \end{frame}
 
+
 \begin{frame}[fragile]
-  \frametitlecpp[11]{Full power of noexcept}
-  \begin{block}{3 uses of noexcept}
+  \frametitlecpp[11]{noexcept}
+  \begin{block}{noexcept}
     \begin{itemize}
-    \item standalone
-      \begin{cppcode*}{gobble=2, linenos=false}
+      \item specifying the \textit{noexcept} specifier states that a function is guaranteed to not throw an exception
+      \begin{cppcode*}{fontsize=\scriptsize,gobble=2,linenos=false}
         int f() noexcept;
       \end{cppcode*}
-    \item as an expression saying whether exceptions can be sent
-      \begin{cppcode*}{gobble=2, linenos=false}
-        int f() noexcept(sizeof(long) == 8);
-      \end{cppcode*}
-    \item as an operator to know whether a function launches exceptions
-      \begin{cppcode*}{gobble=2, linenos=false}
-        constexpr bool callCannotThrow = noexcept(f());
-        if constexpr (callCannotThrow) { ... }
-      \end{cppcode*}
```

Comment:
I believe this usage of noexcept returning a bool has been lost in the refactoring.

Reply:
Yep, it did get lost - I have added it back now